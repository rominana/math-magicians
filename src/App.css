* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

html,
body {
  font-size: 25px;
}

h1 {
  font-size: 30px;
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 500;
  margin: 0 auto;
  position: absolute;
  top: 25%;
  left: 44%;
}
