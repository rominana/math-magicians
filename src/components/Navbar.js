import { Link } from 'react-router-dom';
import '../styles/NavBar.css';

const Navbar = () => (
  <nav className="navbar">
    <h1>
      <Link to="/home" className="logo">
        Math Magicians
      </Link>
    </h1>
    <ul>
      <li>
        <Link to="/home">Home</Link>
      </li>
      <li>
        <Link to="/calculator">Calculator</Link>
      </li>
      <li>
        <Link to="/quote">Quote</Link>
      </li>
    </ul>
  </nav>
);

export default Navbar;
