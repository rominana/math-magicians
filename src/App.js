import React from 'react';
import Calculator from './components/Calculator';
import './App.css';

function App() {
  return (
    <div className="App">
      <h1>React Calculator</h1>
      <Calculator />
    </div>
  );
}

export default App;
