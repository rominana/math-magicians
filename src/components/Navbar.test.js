import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import Navbar from './Navbar';
import App from '../App';

describe('Navbar tests', () => {
  it('Renders Navbar component ', () => {
    const tree = renderer
      .create(
        <Router>
          <Navbar />
          {' '}
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(' navigate to homepage from calculator ', () => {
    render(
      <Router initialEntries={['/calculator']}>
        <App />
        {' '}
      </Router>,
    );
    const homeLink = document.querySelector('#home-link');
    fireEvent.click(homeLink);
    expect(document.body.textContent).toMatch(/welcome/i);
  });

  it('navigate calculator from homepage', () => {
    render(
      <Router initialEntries={['/home']}>
        <App />
        {' '}
      </Router>,
    );
    const calcLink = document.querySelector('#calculator-link');
    fireEvent.click(calcLink);
    expect(document.body.textContent).toMatch(/math/i);
  });

  it('navigate to quote from homepage', () => {
    render(
      <Router initialEntries={['/home']}>
        <App />
        {' '}
      </Router>,
    );
    const quoteLink = document.querySelector('#quote-link');
    fireEvent.click(quoteLink);
    expect(document.body.textContent).toMatch(/hate/i);
  });
});
