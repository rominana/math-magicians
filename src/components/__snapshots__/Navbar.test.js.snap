// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests Renders Navbar component  1`] = `
Array [
  <nav
    className="navbar"
  >
    <h1>
      <a
        className="logo"
        href="/home"
        onClick={[Function]}
      >
        Math Magicians
      </a>
    </h1>
    <ul>
      <li>
        <a
          href="/home"
          id="home-link"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          id="calculator-link"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          id="quote-link"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>,
  " ",
]
`;

exports[`Navbar tests Renders Navbar component correctly 1`] = `
Array [
  <nav
    className="navbar"
  >
    <h1>
      <a
        className="logo"
        href="/home"
        onClick={[Function]}
      >
        Math Magicians
      </a>
    </h1>
    <ul>
      <li>
        <a
          href="/home"
          id="home-link"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          id="calculator-link"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          id="quote-link"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>,
  " ",
]
`;
